### PR TITLE
NCSDECOMP: Smaller fixes

### PR DESCRIPTION
**Pull request by Nostritius**
Two samller things fixing for ncsdecomp

### CODE REVIEW

File: src/nwscript/decompiler.cpp [hunk 1]
```diff
@@ -301,6 +301,11 @@ void Decompiler::writeInstruction(Common::WriteStream &out, const Instruction* i
 		}
 
 		case kOpcodeRSADD: {
+			if (instruction->variables.empty()) {
+				out.writeString("// RSADD not interpretable");
+				break;
+			}
```
Comment (DrMcCoy) [inline src/nwscript/decompiler.cpp]: Wait, how does this even happen?

File: src/nwscript/decompiler.cpp [hunk 2]
```diff
@@ -38,7 +38,7 @@ void Decompiler::createNSS(Common::WriteStream &out) {
 	_ncs->analyzeStack();
 	_ncs->analyzeControlFlow();
 
-	out.writeString("// Decompiled using ncsdecomp");
+	out.writeString("// Decompiled using ncsdecomp\n");
```
Comment (DrMcCoy) [inline src/nwscript/decompiler.cpp]: That should probably be its own commit

Comment (Nostritius) [reply src/nwscript/decompiler.cpp]: Actually, i accidentally inserted this into the commit, i will remove this.

File: src/nwscript/decompiler.cpp (hunk 1 above)
Comment (Nostritius) [reply src/nwscript/decompiler.cpp]: I discovered this in some of the scripts _start functions. A good example is  c_local_set from kotor2

Comment (DrMcCoy) [reply src/nwscript/decompiler.cpp]: The first `RSADDI`?

Hmm, that's the script return value, the `int` in the `int StartingConditional()` xoreos-tools should probably rather create a variable for that.

Comment (DrMcCoy) [reply src/nwscript/decompiler.cpp]: Hmm, wait, or is it?

Comment (DrMcCoy) [reply src/nwscript/decompiler.cpp]: Oh. Looking at it, it seems like yeah, the _start of the script itself will have a return value, not just the main, if it's a StartingConditional.

I.e. that's probably something the stack analysis can do at the last step (or close to). If it detected the script is dialogue conditional script with a StartingConditional (which it already can do), look for the first instruction in _start, and if it's an RSADDI (which it always should be), add a variable there.

But why is it not dectecting that the StartingConditional is copying its return value into there?

@Nostritius, do you feel up to looking into that? I know the script disassembly stuff is a bit of a confusing maze

Comment (Nostritius) [reply src/nwscript/decompiler.cpp]: Yeah, it's something I was hoping to not to do something on it. I can take a look on it, but this will probably take some time. In the meantime I would suggest to change the inserted comment into something like: "// TODO: Add analyzation of RSADD in start".

**Comment by DrMcCoy** [comment]
Okay, merged as 4e319da...3067c50, with your proposed comment change. Thanks! :)
